Foodiea/HomeFeed/Cells/HomeCell.m

```diff
@@ -0,0 +1,131 @@
+        UIImage *img = [UIImage imageNamed:imageName];
+        [self.bookmarkView setImage:img];
+        self.bookmarked = NO;
+    } else {
+        PFUser *user = [PFUser currentUser];
+        PFRelation *relation = [user relationForKey:@"bookmarks"];
+        [relation addObject:self.post];
+        [self.manager saveUserInfo:user];
+        NSString *imageName = @"bookmark-full.png";
+        UIImage *img = [UIImage imageNamed:imageName];
+        [self.bookmarkView setImage:img];
+        self.bookmarked = YES;
+    }
+    
+}
+
+- (void)setPost:(Post *)newPost {
+    //maybe should be _post
+    _post = newPost;
+    //self.postImage.file = post[@"image"];
+    //[self.postImage loadInBackground];
+    self.postCaption.text = self.post[@"caption"];
+    self.usernameLabel.text = self.post.author.username;
+    NSDate *dateVisited = self.post[@"date"];
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    [formatter setDateFormat:@"EEEE, MMMM dd yyyy"];
+
+    //Optionally for time zone conversions
+    [formatter setTimeZone:[NSTimeZone timeZoneWithName:@"..."]];
+
+    NSString *dateString = [formatter stringFromDate:dateVisited];
+
+    self.dateLabel.text = dateString;
+    self.priceLabel.text = self.post[@"price"];
+    self.restaurantName.text = [NSString stringWithFormat:@"%@%@", @"Ate at ",  self.post[@"restaurantName"]];
+    //image
+    self.postImage.file = self.post[@"picture"];
+    [self.postImage loadInBackground];
+    self.userImage.file = self.post.author[@"profileImage"];
+    [self.userImage loadInBackground];
+    
+    //bookmark
+    //goal: see if the user has the post in their bookmark relation
+    // create a relation based on the authors key
+    PFRelation *relation = [[PFUser currentUser] relationForKey:@"bookmarks"];
+    // generate a query based on that relation
+    PFQuery *query = [relation query];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
+        if ([posts count] != 0) {
+            // do something with the array of object returned by the call
+            NSLog(@"%@", posts);
+            for (Post* potential in posts) {
+                NSLog(@"potential");
+                NSLog(@"%@", potential);
+                NSLog(@"%@", potential.objectId);
+                NSLog(@"%@", self.post.objectId);
+                if ([potential.objectId isEqualToString:self.post.objectId]) {
+                    NSLog(@"bookmarked");
+                    NSString *imageName = @"bookmark-full.png";
+                    UIImage *img = [UIImage imageNamed:imageName];
+                    [self.bookmarkView setImage:img];
+                    self.bookmarked = YES;
+                    break;
+                }
+                // do stuff
+            }
+        } else {
+            NSLog(@"no bookmarks");
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
+
+-(BOOL)isBookmarked {
```

Comment:
Let's follow MVC for this project. This code should exist in a model that has a function that call the ApiManger to the this relation and query. 

Essentially, break this up into HomePostView and HomePostModel. See [issue for indepth look](https://github.com/laurajank2/foodiea/issues/2)